### PR TITLE
236 | Correcting syntax error in workflow file

### PR DESCRIPTION
**Pull request by ns-rse**

### CODE REVIEW

File: docs/conf.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
 # Mock imports (things that can't be installed at do building time)
 
 autodoc_mock_imports = ["pygtk", "gwy", "numpy", "pandas", "matplotlib", "seaborn", "scipy", "skimage", "pySPM"]
+import matplotlib
```
Comment (MaxGamill-Sheffield) [inline docs/conf.py:25]: How come this import is standard and not part of the list like above?

**Review by MaxGamill-Sheffield — CHANGES_REQUESTED** [review]
Hey Neil, not quite sure what the error was, nor how it was resolved by all of these changes, but running TS works for me and tests pass fine.

I see the `.rst` files have been edited (not sure what they are), but it looks like all have a name structure such as `<dir>.<module_name>.rst` which is why I'm confused about the removal of the `...dnatracing...` one and addition of `...tracing...`?

File: docs/conf.py (hunk 1 above)
Comment (ns-rse) [reply docs/conf.py:25]: The `autodoc_mock_imports` sets up a list of packages to mock if they are not installed/available when they come to be loaded.

In this instance though installing the package makes `matplotlib` available in the environment and when `grains`, `grainstats` and `run_topostats` made calls to `plot_and_save` it imported `from mpl_toolkits.axes_grid1 import make_axes_locatable` and threw this error thrice over...

```
WARNING: autodoc: failed to import module 'run_topostats' from module 'topostats'; the following exception was raised:
Traceback (most recent call last):
  File "/home/neil/.virtualenvs/topostats-docs/lib/python3.10/site-packages/sphinx/ext/autodoc/importer.py", line 58, in import_module
    return importlib.import_module(modname)
  File "/usr/lib/python3.10/importlib/__init__.py", line 126, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1050, in _gcd_import
  File "<frozen importlib._bootstrap>", line 1027, in _find_and_load
  File "<frozen importlib._bootstrap>", line 1006, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 688, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 883, in exec_module
  File "<frozen importlib._bootstrap>", line 241, in _call_with_frames_removed
  File "/home/neil/work/projects/topostats/TopoStats/topostats/run_topostats.py", line 15, in <module>
    from topostats.grainstats import GrainStats
  File "/home/neil/work/projects/topostats/TopoStats/topostats/grainstats.py", line 15, in <module>
    from topostats.plottingfuncs import plot_and_save
  File "/home/neil/work/projects/topostats/TopoStats/topostats/plottingfuncs.py", line 8, in <module>
    from mpl_toolkits.axes_grid1 import make_axes_locatable
  File "/home/neil/.virtualenvs/topostats-docs/lib/python3.10/site-packages/mpl_toolkits/axes_grid1/__init__.py", line 5, in <module>
    from .parasite_axes import host_subplot, host_axes
  File "/home/neil/.virtualenvs/topostats-docs/lib/python3.10/site-packages/mpl_toolkits/axes_grid1/parasite_axes.py", line 161, in <module>
    with _api.suppress_matplotlib_deprecation_warning():
AttributeError: __enter__
```

The solution I hit upon yesterday was to forcibly import the mocked version of Matplotlib as above _before_ that which is in the environment is used. That seems to get round the error.

Any of the other packages in that list which are not installed in the environment but called when generating documentation using Sphinx will use the mocked version.

**Comment by ns-rse** [comment]
> Hey Neil, not quite sure what the error was, nor how it was resolved by all of these changes, but running TS works for me and tests pass fine.

This is pretty much all to do with docstrings and building the documentation so I shouldn't have touched anything in TopoStats...so its good it works and the tests pass!
 
> I see the `.rst` files have been edited (not sure what they are), but it looks like all have a name structure such as `<dir>.<module_name>.rst` which is why I'm confused about the removal of the `...dnatracing...` one and addition of `...tracing...`?

A bunch of `.rst` files are included so that the modules of the API are listed on the front-page and the sidebar on the left.

There were outdated ones which I've removed (such as `topotracing.rst`) and some needed adjusting because of their location. That is the case with removing `topostats.dnatracing.rst` and adding `topostats.tracing.rst` because the former module doesn't exist its actually the later (see `topostats/tracing/dnatracing.rst`). The "trick" here though is to include the module level of `topostats.tracing.rst` and within that declare the `automodule` to include the correct components, in this case `.. automodule:: topostats.tracing.dnatracing` and hey presto the API documentation for all the functions in `topostats/tracing/dnatracing.py` are included.

I'm not too hot at all this Sphinx stuff so if there is anything you think that could be improved please do let me know.

**Review by MaxGamill-Sheffield — APPROVED** [review]
Looks good! Pages now align with the contents of the files seen in `topostats/` and grainstats is also there.

Note to self: Build html pages via `pip install sphinx` and then pip install all modules it can't find when you run 'sphinx-build docs/ html`. It should put all output files into the `html` dir.

**Pull request merged**